src/store/ducks/opensrp/assignments/index.ts

```diff
@@ -54,14 +55,55 @@ export type ImmutableAssignmentsStoreState = AssignmentsStoreState &
   SeamlessImmutable.ImmutableObject<AssignmentsStoreState>;
 
 /** initial state for Assignments records in store */
-const initialOrgsStoreState: ImmutableAssignmentsStoreState = SeamlessImmutable({
+const initialAssignmentStoreState: ImmutableAssignmentsStoreState = SeamlessImmutable({
   assignmentsByPlanId: {},
 });
 
 /** the Assignment reducer function */
-export default function reducer(state = initialOrgsStoreState, action: AssignmentActionTypes) {
+export default function reducer(
+  state = initialAssignmentStoreState,
+  action: AssignmentActionTypes
+) {
   switch (action.type) {
     case ASSIGNMENTS_FETCHED:
+      if (state.assignmentsByPlanId.asMutable() === {}) {
+        return SeamlessImmutable({
+          ...state,
+          assignmentsByPlanId: action.assignmentsByPlanId,
+        });
+      }
+
+      if (!action.overwrite) {
```

Comment:
`overwrite seem to be missing in the action's interface.`

Reply:
Added this